### PR TITLE
Fix/warning fix

### PR DESCRIPTION
**Pull request by hroest**
fix several warnings reported by clang

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp [hunk 1]
```diff
@@ -185,9 +185,9 @@ namespace OpenMS
     OpenMS::MRMIonSeries mrmis;
 
     size_t j = 0;
-    for (size_t i = 0; i < exp.getTransitions().size(); ++i)
+    for (Size tr_it = 0; tr_it < exp.getTransitions().size(); ++tr_it)
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp]: could you find another name than *_it (like 'j' or 'k')?!
It suggests this is an iterator, which it is not.

File: src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp [hunk 2]
```diff
@@ -537,9 +537,9 @@ namespace OpenMS
     // Go through all peptides and apply the decoy method to the sequence
     // (pseudo-reverse, reverse or shuffle). Then set the peptides and proteins of the decoy
     // experiment.
-    for (Size i = 0; i < exp.getPeptides().size(); i++)
+    for (Size pep_it = 0; pep_it < exp.getPeptides().size(); ++pep_it)
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp]: same here

File: src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp [hunk 3]
```diff
@@ -561,9 +561,9 @@ namespace OpenMS
       {
         peptide = MRMDecoy::shufflePeptide(peptide, identity_threshold, -1, max_attempts);
       }
-      for (Size i = 0; i < peptide.protein_refs.size(); i++)
+      for (Size prot_it = 0; prot_it < peptide.protein_refs.size(); ++prot_it)
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp]: and here

File: src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp [hunk 4]
```diff
@@ -441,17 +441,17 @@ RNPxlModificationMassesResult RNPxlModificationsGenerator::initModificationMasse
 }
 
 //static
-void  RNPxlModificationsGenerator::generateTargetSequences(const String& res_seq, Size pos, const map<char, vector<char> >& map_source2target, StringList& target_sequences)
+void  RNPxlModificationsGenerator::generateTargetSequences(const String& res_seq, Size param_pos, const map<char, vector<char> >& map_source2target, StringList& target_sequences)
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp]: this this is public, I think the test signature needs adaptation too?!

File: src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerCWT.cpp [hunk 5]
```diff
@@ -1477,9 +1477,9 @@ namespace OpenMS
     sort(index_tic_vec.begin(), index_tic_vec.end(), PairComparatorSecondElement<std::pair<Size, double> >());
     
     std::vector<double> best_FWHMs;
-    for (Size i = 0; i < 3 && i < index_tic_vec.size(); ++i)
+    for (Size vec_it = 0; vec_it < 3 && vec_it < index_tic_vec.size(); ++vec_it)
```
Comment (cbielow) [inline src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerCWT.cpp]: same here

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 6]
```diff
@@ -587,6 +587,7 @@ namespace OpenMS
               }
               else if (!swn.second.empty())
               {
+                // TODO: why iterate here if we break in either case ?
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: @mwalzer

File: src/utils/MetaProSIP.cpp [hunk 7]
```diff
@@ -225,7 +225,7 @@ class MetaProSIPInterpolation
       double dxdy = spline.derivatives(xi, 1);
       double yi = spline.eval(xi);
 
-      if (debug >= 100)
+      if (debug)
```
Comment (timosachsenberg) [inline src/utils/MetaProSIP.cpp]: please only output this very verbose mesages for a very large debug level

File: src/utils/OpenSwathWorkflow.cpp [hunk 8]
```diff
@@ -626,31 +626,33 @@ namespace OpenMS
...
+              for (Size cit = 0; cit < tmp_chromatograms.size(); cit++)
               {
                 // Check TIC and remove empty chromatograms (can happen if the
                 // extraction window is outside the mass spectrometric acquisition
                 // window).
-                double tic = std::accumulate(tmp_out[i]->getIntensityArray()->data.begin(),tmp_out[i]->getIntensityArray()->data.end(),0);
-                LOG_DEBUG << "Chromatogram "  << coordinates[i].id << " with size "
-                  << tmp_out[i]->getIntensityArray()->data.size() << " and TIC " << tic  << std::endl;
+                double tic = std::accumulate(tmp_out[cit]->getIntensityArray()->data.begin(),
+                                             tmp_out[cit]->getIntensityArray()->data.end(),0);
```
Comment (timosachsenberg) [inline src/utils/OpenSwathWorkflow.cpp]: , 0

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp (hunk 6 above)
Comment (mwalzer) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: yeah, there should not be a second break in the else brnch

File: src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp (hunk 1 above)
Comment (hroest) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp]: @cbielow ok, I will try to change this. It is just that single letter variables are hard to search and replace, therefore I also dont like `j`

File: src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp (hunk 2 above)
Comment (hroest) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp]: also I believe that naming the variable about regarding to which vector it indexes makes more sense than just `i`, but maybe `pep_idx` makes more sense than `pep_it` - what do you think?

File: src/utils/MetaProSIP.cpp (hunk 7 above)
Comment (hroest) [inline src/utils/MetaProSIP.cpp]: @timosachsenberg `debug` is boolean, so `>= 100` does not make sense

File: src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp (hunk 2 above)
Comment (cbielow) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp]: Pep_idx is fine with me.
For simple loops (not nested ) i'd still prefer i.
But I can totally see your point. 
-- 
Sent via mobile phone.

Am 17. August 2015 22:01:14 MESZ, schrieb Hannes Roest <notifications@github.com>:

File: src/utils/MetaProSIP.cpp (hunk 7 above)
Comment (timosachsenberg) [inline src/utils/MetaProSIP.cpp]: ups I agree :) maybe you could remove the code if you touch the files again (or leave your change as is...)

File: src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp (hunk 4 above)
Comment (hroest) [inline src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp]: turns out there is no test... :-(

Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp]: Yes sorry, tests will follow asap. 
These functions are only used in a util, placed in subfolder and under quite heavy developement.
See e.g.
https://github.com/timosachsenberg/OpenMS/tree/feature/RNPxlSearch_postscore
Hope that somehow justifies my lazyness to create the tests earlier.

Comment (hroest) [inline src/openms/source/ANALYSIS/RNPXL/RNPxlModificationsGenerator.cpp]: no worry - but if there are no tests, no need to adopt the signature :-)

**Comment by cbielow** [comment]
only ppwavelet_3 failed

+1 for merge

**Pull request merged**